.github/workflows/release.yaml

```diff
@@ -0,0 +1,91 @@
+name: Release pypi package
+on:
+  push:
+    branches:
+      - release
+env:
+  PYTHON_VERSION: '3.11'
+jobs:
+  lint:
+    name: Run linters
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@b4ffde65f46336ab88eb53be808477a3936bae11 # v4
+      - uses: actions/setup-python@v4
+        with:
+          python-version: ${{ env.PYTHON_VERSION }}
+      - uses: snok/install-poetry@v1
+        with:
+          version: 1.6.1
+          virtualenvs-create: false
+      - name: Install dependencies and run lints
+        run: |
+          # Necessary for running the mypy pre-commit hook
+          poetry install
+      - uses: pre-commit/action@v3.0.0
+  test:
+    name: Run tests
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@b4ffde65f46336ab88eb53be808477a3936bae11 # v4
+      - uses: actions/setup-python@v4
+        with:
+          python-version: ${{ env.PYTHON_VERSION }}
+      - uses: snok/install-poetry@v1
+        with:
+          version: 1.6.1
+          virtualenvs-create: false
+      - name: Install core dependencies
+        run: poetry install
+      - name: Run pytest
+        run: pytest tests
+  build:
+    runs-on: ubuntu-latest
+    environment: dev
+    env:
+      CDF_CLUSTER: 'greenfield'
+      CDF_PROJECT: 'project-loader-dev'
+    name: Test build templates
+    steps:
+      - uses: actions/checkout@v4
+      - uses: actions/setup-python@v4
+        with:
+          python-version: 3.11
+      - uses: snok/install-poetry@v1
+        with:
+          version: 1.6.1
+          virtualenvs-create: false
+      - name: Install dependencies and build the package
+        run: |
+          poetry install
+          poetry build
+      - name: Install cdf-tk
+        run: pip install .
+      - name: Initialize project
+        run: cdf-tk init demo_project
+      - name: "Pre-processing for demo environment"
+        run: ./demo/preproc.sh
+      - name: "Build the templates"
+        run: cdf-tk build --build-dir=./build --env=demo ./demo_project
+  deploy:
+    runs-on: ubuntu-latest
+    environment: CD
+    needs: [lint, test, build]
```

Comment:
This has been a debate in the Python-SDK. Do we run everything of linting, test and build in main before release? It has already been run on the branch before merging. The problem in the SDK is that sometimes due to instability in the API (very often Transformations) the tests fail and the release does not do through. This leads to multiple versions never being released to pypi.

Reply:
Yes, see that point. But we are not doing anything live here. With a release branch, you can theoretically merge in from something else than main (unless we do extra enforcement), so this is an extra safeguard.